### PR TITLE
Override all the labels of the pattern categories taxonomy

### LINKED ISSUE #57076: Pattern categories referenced as tags in site editor post taxonomies panel
Looks like #57049 has led to the pattern categories UI to suggest these are "tags", though they are categories. Perhaps this panel should be updated to use the existing categories UI anyhow; instead of the FormFieldToken, which is used for tags on posts. 

As an aside, I think it'd be fine to just have "Categories" as the panel title, as we do the same on the post editor (and the UI already suggests this is a custom post type via the tab). 

cc @youknowriad

CleanShot 2023-12-14 at 13 08 01
CleanShot 2023-12-14 at 13 07 20

### PR DESCRIPTION
**Pull request by youknowriad**
closes #57076 

## What?

The site editor has been updated to use the normal post taxonomies component, this highlighted the fact that the "pattern categories" taxonomy didn't have all the labels defined properly. This PR fixes that.

## Testing Instructions

 - Open a synced pattern in the site editor
 - Notice the "pattern categories" panel on the editor sidebar
 - It says "Add pattern category" rather than "Add tag"

### CODE REVIEW

File: lib/compat/wordpress-6.5/block-patterns.php [hunk 1]
```diff
@@ -31,3 +31,50 @@ function gutenberg_register_media_pattern_categories() {
...
+ */
+function gutenberg_register_taxonomy_patterns() {
+	$args = array(
+		'public'             => false,
+		'publicly_queryable' => false,
+		'hierarchical'       => false,
+		'labels'             => array(
+			'name'                       => _x( 'Pattern Categories', 'taxonomy general name' ),
+			'singular_name'              => _x( 'Pattern Category', 'taxonomy singular name' ),
+			'add_new_item'               => __( 'Add New Pattern Category' ),
```
Comment (richtabor) [inline lib/compat/wordpress-6.5/block-patterns.php]: What do you think about making this "Add new category" instead? It's a bit duplicative as-is: 

CleanShot 2023-12-15 at 17 14 36

File: lib/compat/wordpress-6.4/block-patterns.php
Comment (richtabor) [inline lib/compat/wordpress-6.4/block-patterns.php:1]: Just double-checking it's ok to remove this?

Comment (youknowriad) [reply lib/compat/wordpress-6.4/block-patterns.php:1]: Yep, it's fine, it was moved.

File: lib/compat/wordpress-6.5/block-patterns.php [hunk 2]
```diff
@@ -31,3 +31,50 @@ function gutenberg_register_media_pattern_categories() {
 	);
 }
 add_action( 'init', 'gutenberg_register_media_pattern_categories' );
+
+/**
+ * Adds a new taxonomy for organizing user created patterns.
+ *
+ * Note: This should be removed when the minimum required WP version is >= 6.4.
+ *
+ * @see https://github.com/WordPress/gutenberg/pull/53163
```
Comment (ntsekouras) [inline lib/compat/wordpress-6.5/block-patterns.php]:
```suggestion
 *
 * @see https://github.com/WordPress/gutenberg/pull/53163
```

**Review by ntsekouras — APPROVED** [review]
Looks good, thanks!

**Pull request merged**